addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechOptionsPanel.java

```diff
@@ -0,0 +1,124 @@
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.TitledBorder;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.OptionsParam;
+import org.parosproxy.paros.view.AbstractParamPanel;
+import org.zaproxy.zap.extension.wappalyzer.ExtensionWappalyzer.Mode;
+import org.zaproxy.zap.utils.FontUtils;
+
+public class TechOptionsPanel extends AbstractParamPanel {
+
+    private static final long serialVersionUID = -4195576861254405033L;
+
+    private static final String NAME = Constant.messages.getString("wappalyzer.optionspanel.name");
+    private static final String NAME_MODE =
+            Constant.messages.getString("wappalyzer.optionspanel.mode");
+
+    private JComboBox<Mode> modeComboBox;
+    private JPanel modePanel;
+
+    public TechOptionsPanel() {
+        super();
+        setName(NAME);
+
+        setLayout(new BorderLayout(0, 0));
+
+        final JPanel panel = new JPanel(new GridBagLayout());
+        panel.setBorder(new EmptyBorder(2, 2, 2, 2));
+
+        GridBagConstraints gbc = new GridBagConstraints();
+        gbc.gridx = 0;
+        gbc.gridy = 0;
+        gbc.insets = new java.awt.Insets(2, 2, 2, 2);
+        gbc.anchor = GridBagConstraints.NORTH;
+        gbc.fill = GridBagConstraints.HORIZONTAL;
+        gbc.weighty = 0.0D;
+        gbc.weightx = 1.0D;
+        panel.add(getModePanel(), gbc);
+
+        add(panel, BorderLayout.NORTH);
+    }
+
+    private JPanel getModePanel() {
+        if (modePanel == null) {
+            modePanel = new JPanel();
+            modePanel.setLayout(new GridBagLayout());
+
+            GridBagConstraints gbc = new GridBagConstraints();
+
+            modePanel.setBorder(
+                    BorderFactory.createTitledBorder(
+                            null,
+                            NAME_MODE,
+                            TitledBorder.DEFAULT_JUSTIFICATION,
+                            TitledBorder.DEFAULT_POSITION,
+                            FontUtils.getFont(FontUtils.Size.standard)));
+
+            gbc.gridx = 0;
+            gbc.gridy = 0;
+            gbc.insets = new java.awt.Insets(2, 2, 2, 2);
+            gbc.anchor = GridBagConstraints.WEST;
+            gbc.fill = GridBagConstraints.HORIZONTAL;
+            gbc.weightx = 0.5D;
+            modePanel.add(new JLabel(NAME_MODE), gbc);
+
+            gbc.gridx = 1;
+            gbc.gridy = 0;
+            gbc.ipadx = 50;
+            modePanel.add(getModeComboBox(), gbc);
+        }
+        return modePanel;
+    }
+
+    private JComboBox<Mode> getModeComboBox() {
+        if (modeComboBox == null) {
+            modeComboBox = new JComboBox<>(new DefaultComboBoxModel<>(Mode.values()));
+        }
+        return modeComboBox;
+    }
+
+    @Override
+    public void initParam(Object obj) {
+        final OptionsParam options = (OptionsParam) obj;
+        final WappalyzerParam param = options.getParamSet(WappalyzerParam.class);
```

Comment:
finals could be dropped (more places we don't use than the ones that use).

Reply:
Okay, I think that was a c/p from encoder